src/pkg/control/restore.go

```diff
@@ -0,0 +1,86 @@
+package control
+
+import (
+	"context"
+	"strings"
+
+	"golang.org/x/exp/slices"
+
+	"github.com/alcionai/corso/src/internal/common/dttm"
+	"github.com/alcionai/corso/src/pkg/logger"
+)
+
+const (
+	defaultRestoreLocation = "Corso_Restore_"
+)
+
+// CollisionPolicy describes how the datalayer behaves in case of a collision.
+type CollisionPolicy string
+
+const (
+	Unknown CollisionPolicy = ""
+	Skip    CollisionPolicy = "skip"
+	Copy    CollisionPolicy = "copy"
+	Replace CollisionPolicy = "replace"
+)
+
+func ValidCollisionPolicies() map[CollisionPolicy]struct{} {
+	return map[CollisionPolicy]struct{}{
+		Skip:    {},
+		Copy:    {},
+		Replace: {},
+	}
+}
+
+const RootLocation = "/"
+
+// RestoreConfig contains
+type RestoreConfig struct {
+	// Defines the per-item collision handling policy.
+	// Defaults to Skip.
+	OnCollision CollisionPolicy
+
+	// ProtectedResource specifies which resource the data will be restored to.
+	// If empty, restores to the same resource that was backed up.
+	// Defaults to empty.
+	ProtectedResource string
+
+	// Location specifies the container into which the data will be restored.
+	// Only accepts container names, does not accept IDs.
+	// If empty or "/", data will get restored in place, beginning at the root.
+	// Defaults to "Corso_Restore_<current_dttm>"
+	Location string
+
+	// Drive specifies the drive into which the data will be restored.
+	// If empty, data is restored to the same drive that was backed up.
+	// Defaults to empty.
+	Drive string
+}
+
+func DefaultRestoreConfig(timeFormat dttm.TimeFormat) RestoreConfig {
+	return RestoreConfig{
+		OnCollision: Skip,
+		Location:    defaultRestoreLocation + dttm.FormatNow(timeFormat),
+	}
+}
+
+// EnsureRestoreConfigDefaults sets all non-supported values in the config
+// struct to the default value.
+func EnsureRestoreConfigDefaults(
+	ctx context.Context,
+	rc RestoreConfig,
+) RestoreConfig {
+	if !slices.Contains([]CollisionPolicy{Skip, Copy, Replace}, rc.OnCollision) {
+		logger.Ctx(ctx).
+			With(
+				"bad_collision_policy", rc.OnCollision,
+				"default_collision_policy", Skip).
+			Info("setting collision policy to default")
+
+		rc.OnCollision = Skip
```

Comment:
I'm a bit on the fence about this. This PR can go through as is, but would be good to check with others about the behavior. I'm concerned that not returning an error here could lead to unexpected results and difficulty tracking down bugs because everything seems to be working

Reply:
I'm not so concerned, because the only way this'll happen is if: 1. we introduce a new policy (highly unlikely) but forget to add it to the valid set; or 2, if the caller sets the value to a typecast string instead of an enum (also unlikely).  If this were an arbitrary string, instead of a typed enum, I'd have more concern over it.

Reply:
forgetting to set it in the struct is also an option. In which case things will work, but may not be what's expected and finding the issue could be difficult

Reply:
`Forgetting` is always a toss-up for me, since it overlaps conceptually with `intentionally unset`, and I don't recall that we've ever standardized our configuration around required population; opting instead for falling back to defaults.